cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MigrationTableUtility.java

```diff
@@ -19,80 +19,60 @@
 import co.cask.cdap.data2.util.TableId;
+import co.cask.cdap.data2.util.hbase.HBaseDDLExecutorFactory;
 import co.cask.cdap.data2.util.hbase.HBaseTableUtil;
 import co.cask.cdap.metrics.store.MetricDatasetFactory;
 import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.spi.hbase.HBaseDDLExecutor;
 import org.apache.hadoop.conf.Configuration;
-import org.apache.hadoop.hbase.TableNotFoundException;
 import org.apache.hadoop.hbase.client.HBaseAdmin;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
-import java.util.concurrent.TimeUnit;
 import javax.annotation.Nullable;
 
 /**
- * Useful to get v2/v3 metrics tables with retry and check if metrics table exists or not
+ * To get v2/v3 metrics tables; perform tableExists/delete table on v2 metrics tables
  */
 public class MigrationTableUtility {
   private static final Logger LOG = LoggerFactory.getLogger(MigrationTableUtility.class);
 
   private final CConfiguration cConf;
   private final Configuration hConf;
   private final MetricDatasetFactory metricDatasetFactory;
-  private final int resolution;
+  private final HBaseDDLExecutorFactory hBaseDDLExecutorFactory;
   private final HBaseTableUtil tableUtil;
 
   MigrationTableUtility(CConfiguration cConfiguration, Configuration hConf,
-                        MetricDatasetFactory metricDatasetFactory, HBaseTableUtil tableUtil, int resolution) {
+                        MetricDatasetFactory metricDatasetFactory, HBaseTableUtil tableUtil) {
     this.cConf = cConfiguration;
     this.hConf = hConf;
     this.metricDatasetFactory = metricDatasetFactory;
+    this.hBaseDDLExecutorFactory = new HBaseDDLExecutorFactory(cConf, hConf);
     this.tableUtil = tableUtil;
-    this.resolution = resolution;
   }
 
-  // we try to get v2 metrics table with retry, there is a chance that v2 table got deleted before we got it
-  // we return null if it was already deleted
+  // we try to get v2 metrics table; if there is any exception while getting the table we return null
   @Nullable
-   MetricsTable getV2MetricsTable() {
+  MetricsTable getV2MetricsTable(int resolution) {
     MetricsTable v2table = null;
-    while (v2table == null) {
-      try {
-        v2table = metricDatasetFactory.getV2MetricsTable(resolution);
-      } catch (Exception e) {
-        if (e instanceof TableNotFoundException) {
-          // we checked table exists, however the table got deleted in-between before we got the table
-          return null;
-        }
-        LOG.warn("Cannot access v2 metricsTable, will retry in 1 sec.");
-        try {
-          TimeUnit.SECONDS.sleep(1);
-        } catch (InterruptedException ie) {
-          Thread.currentThread().interrupt();
-          break;
-        }
-      }
+    try {
+      v2table = metricDatasetFactory.getV2MetricsTable(resolution);
+    } catch (Exception e) {
+      LOG.debug("Cannot access v2 metricsTable due to exception", e);
```

Comment:
maybe this should be a warning or at least info. It prevents migration, so it must be surfaced. Also, if the table does not exist, don't you want to handle that differently?